modules/http/http.c

```diff
@@ -491,6 +223,324 @@ http_dd_set_timeout(LogDriver *d, glong timeout)
+       * headers and it makes batching several messages into the same request a
+       * bit more complicated than it needs to be.  I didn't want to break
+       * backward compatibility when batching was introduced, however I think
+       * this should eventually be removed */
+
+      headers = _add_header(headers,
+                            "X-Syslog-Host",
+                            log_msg_get_value(msg, LM_V_HOST, NULL));
+      headers = _add_header(headers,
+                            "X-Syslog-Program",
+                            log_msg_get_value(msg, LM_V_PROGRAM, NULL));
+      headers = _add_header(headers,
+                            "X-Syslog-Facility",
+                            syslog_name_lookup_name_by_value(msg->pri & LOG_FACMASK, sl_facilities));
+      headers = _add_header(headers,
+                            "X-Syslog-Level",
+                            syslog_name_lookup_name_by_value(msg->pri & LOG_PRIMASK, sl_levels));
+    }
+
+  for (l = self->headers; l; l = l->next)
+    headers = curl_slist_append(headers, l->data);
+
+  return headers;
+}
+
+static void
+_add_message_to_batch(HTTPDestinationDriver *self, LogMessage *msg)
+{
+  if (self->body_template)
+    {
+      log_template_append_format(self->body_template, msg, &self->template_options, LTZ_SEND,
+                                 self->super.seq_num, NULL, self->request_body);
+    }
+  else
+    {
+      if (self->request_body->len)
+        g_string_append_c(self->request_body, '\n');
+      g_string_append(self->request_body, log_msg_get_value(msg, LM_V_MESSAGE, NULL));
+    }
+}
+
+static worker_insert_result_t
+_map_http_status_to_worker_status(glong http_code)
+{
+  worker_insert_result_t retval;
+
+  switch (http_code/100)
+    {
+    case 4:
+      retval = WORKER_INSERT_RESULT_DROP;
+      break;
+    case 5:
+      retval = WORKER_INSERT_RESULT_ERROR;
+      break;
+    default:
+      retval = WORKER_INSERT_RESULT_SUCCESS;
+      break;
+    }
+
+  return retval;
+}
+
+/* we flush the accumulated data if
+ *   1) we reach batch_size,
+ *   2) the message queue becomes empty
+ */
+static worker_insert_result_t
+_flush(LogThreadedDestDriver *s)
+{
+  HTTPDestinationDriver *self = (HTTPDestinationDriver *) s;
+  CURLcode ret;
+  worker_insert_result_t retval;
+
+  if (self->super.batch_size == 0)
+    return WORKER_INSERT_RESULT_SUCCESS;
+
+  curl_easy_setopt(self->curl, CURLOPT_HTTPHEADER, self->request_headers);
+  curl_easy_setopt(self->curl, CURLOPT_POSTFIELDS, self->request_body->str);
+
+  if ((ret = curl_easy_perform(self->curl)) != CURLE_OK)
+    {
+      msg_error("curl: error sending HTTP request",
+                evt_tag_str("error", curl_easy_strerror(ret)),
+                log_pipe_location_tag(&self->super.super.super.super));
+      retval = WORKER_INSERT_RESULT_NOT_CONNECTED;
+      goto exit;
+    }
+
+  glong http_code = 0;
+
+  curl_easy_getinfo(self->curl, CURLINFO_RESPONSE_CODE, &http_code);
```

Comment:
According to the documentation the `http_code` is filled only if the function returns with `CURLE_OK`, and the current code initialize `http_code` to `0`, and later divides it with `100` and if it is not `4` or `5` it is success.
It could report unsuccessful calls as successful one. (I know it has a low chance, as this is just getting the information, but still.)